server.go

```diff
@@ -423,12 +429,16 @@ func (srv *Server) Shutdown() error {
 		return &Error{err: "server not started"}
 	}
 	srv.started = false
-	srv.lock.Unlock()
+	defer srv.lock.Unlock() // to make sure re-serve works do after shutdown
```

Comment:
if remove defer, can't make sure a server will not start during it's shutting down. but reuse srv.lock instead of shutdownSignal we should use this lock